packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts

```diff
@@ -31,18 +47,38 @@ export default util.createRule<[], MessageIds>({
         'This expression unnecessarily compares a boolean value to a boolean instead of using it directly.',
       negated:
         'This expression unnecessarily compares a boolean value to a boolean instead of negating it.',
+      comparingNullableToTrueDirect:
+        'This expression unnecessarily compares a nullable boolean value to true instead of using it directly.',
+      comparingNullableToTrueNegated:
+        'This expression unnecessarily compares a nullable boolean value to true instead of negating it.',
+      comparingNullableToFalse:
+        'This expression unnecessarily compares a nullable boolean value to false instead of using the ?? operator to provide a default.',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowComparingNullableBooleans: {
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
     type: 'suggestion',
   },
-  defaultOptions: [],
-  create(context) {
+  defaultOptions: [
+    {
+      allowComparingNullableBooleans: true,
```

Comment:
for backwards compatibility, please set this to `false` (and update the docs).
I don't think that this is something everyone will want to have on for this rule - as many people like strict boolean expressions.
```suggestion
      allowComparingNullableBooleans: false,
```

Reply:
Just committed these before realizing that the default before (`allowComparingNullableBooleans=true`) was backwards compatible, and switching it to `false` is no longer backwards compatible.

`allowComparingNullableBooleans=true` means "comparing literal booleans to nullable booleans is allowed so eslint should not complain if I do it" and `allowComparingNullableBooleans=false` means "comparing literal booleans to nullable booleans is not allowed so eslint should complain if I do it".

Happy to change the naming since clearly it's a bit confusing :)